bip-musig2.mediawiki

```diff
@@ -313,17 +314,26 @@ Input:
 ==== Nonce Generation ====
 
 Input:
-* The secret signing key ''sk'': a 32-byte array or 0-byte array (optional argument)
-* The aggregate public key ''aggpk'': a 32-byte array or 0-byte array (optional argument)
-* The message ''m'': a 32-byte array or 0-byte array (optional argument)
+* The secret signing key ''sk'': a 32-byte array (optional argument)
+* The aggregate public key ''aggpk'': a 32-byte array (optional argument)
+* The message ''m'': a byte array with ''0 &le; len(m) &le; 2<sup>32</sup>-1'' (optional argument)
 * The auxiliary input ''extra_in'': a byte array with ''0 &le; len(extra_in) &le; 2<sup>32</sup>-1'' (optional argument)
 
 '''''NonceGen(sk, aggpk, m, extra_in)''''':
 * Let ''rand' '' be a 32-byte array freshly drawn uniformly at random
-* If ''len(sk) > 0'':
+* If the optional argument ''sk'' is present:
 ** Let ''rand'' be the byte-wise xor of ''sk'' and ''hash<sub>MuSig/aux</sub>(rand')''<ref>The random data is hashed (with a unique tag) as a precaution against situations where the randomness may be correlated with the secret signing key itself. It is xored with the secret key (rather than combined with it in a hash) to reduce the number of operations exposed to the actual secret key.</ref>.
 * Else: let ''rand = rand' ''
-* Let ''k<sub>i</sub> = int(hash<sub>MuSig/nonce</sub>(rand || bytes(1, len(aggpk)) || aggpk || bytes(1, len(m)) || m || bytes(4, len(extra_in)) || extra_in || bytes(1, i - 1))) mod n'' for ''i = 1,2''
+* If the optional argument ''aggpk'' is not present:
+** Let ''aggpk = empty_bytestring''
+* If the optional argument ''m'' is not present:
```

Comment:
You want `bytes(1, 0) || bytes(8, 0) || empty_bytestring` in the case where the optional argument `m` was not provided, as opposed to, say, just `bytes(1, 0)`?

I'm ok with this if you are, I just want to confirm.

Reply:
This seemed to result in the simplest spec. But perhaps not if we replace `m_present` entirely. For example, we could do:
```
    * If the optional argument ''m'' is not present:
    ** Let ''m_prefixed = bytes(1, 0)''
    * Else:
    ** Let ''m_prefixed = bytes(1, 1) || || bytes(8, len(m)) || m''
```
And then only hash `m_prefixed`. I have a slight preference for this.

Reply:
Suggestion LGTM